tests/tools/onert_train/src/rawdataloader.cc

```diff
@@ -39,64 +40,76 @@ uint64_t getRawTensorSize(const std::vector<nnfw_tensorinfo> &infos)
 
 namespace onert_train
 {
-std::tuple<Generator, uint32_t>
-RawDataLoader::loadData(const std::string &input_file, const std::string &expected_file,
-                        const std::vector<nnfw_tensorinfo> &input_infos,
-                        const std::vector<nnfw_tensorinfo> &expected_infos,
-                        const uint32_t batch_size)
+
+RawDataLoader::RawDataLoader(const std::string &input_file, const std::string &expected_file,
+                             const std::vector<nnfw_tensorinfo> &input_infos,
+                             const std::vector<nnfw_tensorinfo> &expected_infos)
+  : DataLoader(input_infos, expected_infos)
 {
   _input_file = std::ifstream(input_file, std::ios::binary);
   _expected_file = std::ifstream(expected_file, std::ios::binary);
 
   _input_file.seekg(0, std::ios::end);
   uint32_t input_file_size = _input_file.tellg();
-  uint32_t input_data_length = input_file_size / getRawTensorSize(input_infos);
+  uint32_t input_data_length = input_file_size / getRawTensorSize(_input_infos);
 
   _expected_file.seekg(0, std::ios::end);
   uint32_t expected_file_size = _expected_file.tellg();
-  uint32_t expected_data_length = expected_file_size / getRawTensorSize(expected_infos);
+  uint32_t expected_data_length = expected_file_size / getRawTensorSize(_expected_infos);
 
   if (input_data_length != expected_data_length)
   {
     throw std::runtime_error("The length of input data and expected data does not match.");
   }
 
-  uint32_t data_length = input_data_length;
+  _data_length = input_data_length;
+}
+
+std::tuple<Generator, uint32_t> RawDataLoader::loadData(const uint32_t batch_size, const float from,
+                                                        const float to)
+{
+  assert(from >= 0.f && from <= 1.f);
+  assert(to >= 0.f && to <= 1.f);
+  assert(from <= to);
```

Comment:
```suggestion
  assert(from >= 0.f);
  assert(from <= to);
  assert(to <= 1.f);
```
(optional) Some conditions can be removed.

Reply:
Thank you for suggestion. I will apply it to another pr.